src/main/java/Drew.java

```diff
@@ -0,0 +1,309 @@
+        case DEADLINE:
+            if (inputNoWhitespaceLength == 8) {
+                throw new InsufficientArgumentsException("'Deadline task' cannot be empty");
+            }
+            break;
+        case EVENT:
+            if (inputNoWhitespaceLength == 5) {
+                throw new InsufficientArgumentsException("'Event task' cannot be empty");
+            }
+            break;
+        }
+        return userCommand;
+    }
+    public static Task parseFromSave(String input) {
+        String[] args = input.split("\\|");
+        for (int i = 0; i < args.length; i ++) {
+            args[i] = args[i].trim();
+        }
+        Task task = null;
+        System.out.println(args[0]);
+        switch (args[0]) {
+        case "T":
+            task = new Todo(args[2]);
+            break;
+        case "D":
+            task = new Deadline(args[2], LocalDate.parse(args[3]));
+            break;
+        case "E":
+            task = new Event(args[2], LocalDate.parse(args[3]), LocalDate.parse(args[4]));
+            break;
+        }
+
+        if (args[1] == "1") {
+            task.setDone();
+        }
+        return task;
+    }
+    public static void readFile(String filePath, ArrayList<Task> ls) {
+        File savedTasks = new File(filePath);
+        System.out.println(savedTasks.getAbsolutePath());
+
+        try {
+            Scanner fileReader = new Scanner(savedTasks);
+            System.out.println("Status: File found");
+            while (fileReader.hasNext()) {
+                String line = fileReader.nextLine();
+                System.out.println(line);
+                Task task = parseFromSave(line);
+                System.out.println(task.toStatusString());
+                ls.add(task);
+            }
+        } catch (FileNotFoundException e){
+            System.out.println("Status: File not found");
+        }
+    }
+
+    public static boolean saveFile(String filePath, ArrayList<Task> ls) {
+        File savedTasks = new File(filePath);
+        try {
+            boolean mkdirSuccess = savedTasks.getParentFile().mkdirs();
+            boolean createFileSuccess = savedTasks.createNewFile();
+            System.out.println("mkdir: " + mkdirSuccess);
+            System.out.println("File creation: " + createFileSuccess);
+            //write to file here
+            FileWriter fw = new FileWriter(filePath);
+            int listLength = ls.size();
+            for (int i = 0; i < listLength; i++) {
+                String task = ls.get(i).toSaveFormatString();
+                fw.write(task);
+            }
+            fw.close();
+            return true;
+        } catch (SecurityException e){
+            System.out.println("Error: Privilege");
+            return false;
+        } catch (IOException e) {
+            System.out.println("Error: IO");
+            return false;
+        }
+    }
+
+    public static String executeCommand(ArrayList<Task> ls, String input) {
```

Comment:
Avoid Long Methods, you can consider calling another method to process your different userCommands